test.py

```diff
@@ -1338,29 +1338,67 @@ def format_unidiff(fromfile: str, tofile: str) -> str:
         return buf.getvalue()
 
 
+def update_junit_suite(suite, test):
+    tests = int(suite.get('tests', '0'))
+    failures = int(suite.get('failures', '0'))
+    duration = float(suite.get('time', '0'))
+
+    tests += 1
+    if not test.success:
+        failures += 1
+    duration += test.time_end - test.time_start
+
+    suite.set('tests', f'{tests}')
+    suite.set('failures', f'{failures}')
+    suite.set('time', f'{duration:.3f}')
+
+
 def write_junit_report(tmpdir: str, mode: str) -> None:
+    tests = filter(lambda test: test.mode == mode,
+                   itertools.chain.from_iterable(suite.junit_tests()
+                                                 for suite in TestSuite.suites.values()))
+
+    testsuites = ET.Element("testsuites")
+    for test in tests:
+        # normalize the name, as Java class name does not contain '-'
+        suite_name = test.suite.name.replace('-', '_')
+        suite = testsuites.find(f"./testsuite[@name='{suite_name}']")
+        if suite is None:
+            suite = ET.SubElement(testsuites, 'testsuite',
+                                  name=suite_name,
+                                  error='0')
+        # please note, as we run the tests on the per-file basis, i.e., execute
+        # "run", or execute "pytest <a-file-name-found-by-glob>", we cannot tell
+        # one test from another, so misuse the "classname" for the shortname,
```

Comment:
Please use a consistent terminology in the comment, otherwise I can't parse it.

There is a test suite. It consists of test files. It consists of test cases. So what do you mean actually in this comment?